pkg/runctl/run_manager.go

```diff
@@ -480,9 +477,14 @@ func (c *runManager) createTektonTaskRun(ctx context.Context, runCtx *runContext
 	namespace := runCtx.runNamespace
 	serviceAccountSecretName, err := c.getServiceAccountSecretName(ctx, runCtx)
 	if err != nil {
-		return err
+		return nil, err
+	}
+	log.Printf("Pipeline Run: %+v", runCtx.pipelineRun.GetSpec())
+	timeout := runCtx.pipelineRunsConfig.Timeout
+	pipelineRunTimeout := runCtx.pipelineRun.GetSpec().Timeout
+	if pipelineRunTimeout != nil {
+		timeout = pipelineRunTimeout
 	}
```

Comment:
You may encapsulate this selection logic into a function to separate it from the main code that just assembles the object:

```suggestion
	timeout := func() metav1.Duration {
		timeout := runCtx.pipelineRun.GetSpec().Timeout
		if timeout == nil {
			timeout = &runCtx.pipelineRunsConfig.Timeout
		}
		return *timeout
	}
```

Reply:
Moved fetching of timeout into a separate function.